modules/standard/BigInteger.chpl

```diff
@@ -2906,10 +2917,24 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
       ret = mpz_perfect_power_p(t_.mpz);
     }
 
-    return ret.safeCast(int);
+    if ret.safeCast(int) then 
```

Comment:
Ditto comment in your other PR about checking the value of ret rather than if it can safeCast to an int